lint.go

```diff
@@ -1436,6 +1437,47 @@ func (f *file) lintTimeNames() {
 	})
 }
 
+// lintContextKeyTypes checks for call expressions to context.WithValue with
+// basic types used for the key parameter.
+// See: https://golang.org/issue/17293
+func (f *file) lintContextKeyTypes() {
+	f.walk(func(node ast.Node) bool {
+		switch node := node.(type) {
+		case *ast.CallExpr:
+			f.checkContextKeyType(node)
+		}
+
+		return true
+	})
+}
+
+// checkContextKeyType checks a single call expression to determine if it
```

Comment:
"checkContextKeyType reports an error if the call expression calls context.WithValue with a key argument of basic type."